### PR TITLE
feat: add useGitClone option to lookup table

### PR DESCRIPTION
**Pull request by targos**
If "useGitClone" is set to "true", a shallow clone of the
repository will be used instead of downloading the tarball.

Fixes: https://github.com/nodejs/citgm/issues/694

### CODE REVIEW

**Comment by targos** [comment]
tests are passing now

File: lib/git-clone.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+  const modulePath = path.join(context.path, context.module.name);
+  mkdirp.sync(modulePath);
+  const execGit = getExecGit(modulePath);
+
+  try {
+    execGit(['init']);
+    execGit(['remote', 'add', 'origin', gitUrl]);
+    execGit(['fetch', '--depth=1', 'origin', gitRef]);
+    execGit(['checkout', 'FETCH_HEAD']);
```
Comment (vweevers) [inline lib/git-clone.js]: Alternatively: `checkout --recurse-submodules FETCH_HEAD` (instead of `submodule init` and `update`)

Comment (targos) [reply lib/git-clone.js]: oh nice, I didn't know about this option 👍

Comment (targos) [reply lib/git-clone.js]: Actually, it doesn't work. The git manual says "Using --recurse-submodules will update the content of all initialized submodules according to the commit recorded in the superproject", so it seems we have to `init`. Then it's easier to update.

**Comment by targos** [comment]
CI: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-continuous-integration-pipeline/87/console

**Comment by targos** [comment]
Testing with `leveldown`: https://ci.nodejs.org/view/Node.js-citgm/job/citgm-smoker-pipeline/66/console

File: lib/git-clone.js (hunk 1 above)
Comment (vweevers) [reply lib/git-clone.js]: My bad, I assumed it meant the same as in `git clone --recurse-submodules`.

**Comment by MylesBorins** [comment]
So it was originally a design contraint within citgm not to support git clones, and more specifically not supporting git sub modules.

Are we sure that we want to expand the scope to support this?

**Comment by vweevers** [comment]
An alternative would be for `leveldown` to include tests in its npm package (we're OK with that). Then CITGM can fetch a tarball from npm. This tarball includes everything else already and does not require git.

**Comment by vweevers** [comment]
That's relatively easy to achieve. Make this do `npm pack leveldown` instead of `npm pack <url>`:

https://github.com/nodejs/citgm/blob/c9481e07d7e3dba1a5eb9e2cbe0993b4a19627c6/lib/grab-project.js#L30

Caveat: it will use prebuilt (N-API) binaries unless CITGM installs it with `npm i --build-from-source`.

**Comment by targos** [comment]
I'd like to know why it's a design decision to not support git clones. It is not very much different from downloading the tarball from github and could potentially support projects that are hosted elsewhere

**Comment by BridgeAR** [comment]
@MylesBorins 
> So it was originally a design contraint within citgm not to support git clones, and more specifically not supporting git sub modules.

Are there any downsides to include this feature? Having this possibility just expands the number of modules we are able to include in CITGM and that is IMHO something positive.

**Comment by MylesBorins** [comment]
TBH it was about encouraging practices to allow the publised modules to npm to be able to be tested... it is very reasonable to step away from this goal as it could be considered fairly rigid

**Comment by BridgeAR** [comment]
@targos I guess this is ready to land?

**Pull request merged**